node/chainSimulator/configs/configs.go

```diff
@@ -181,14 +184,34 @@ func updateSupernovaConfigs(configs *config.Configs, args ArgsChainSimulatorConf
 		}
 	}
 
-	if args.RoundsPerEpoch.HasValue {
-		// update supernova round for the new rounds per epoch config
-		newRoundsPerEpoch := args.RoundsPerEpoch.Value
-		newSupernovaRound := uint64(supernovaEpoch)*newRoundsPerEpoch + 5 // 5 rounds later
-		if isSupernovaFromGenesis {
-			// if supernova is from genesis, the round should be 0 as well
-			newSupernovaRound = 0
+	if !args.RoundsPerEpoch.HasValue {
+		return
+	}
+
+	hasCorrectActionRound := false
+	diff := int(supernovaEpoch) - int(args.InitialEpoch)
+	if diff >= 0 {
+
+		correctRoundActivationForSupernova := args.InitialRound + int64(diff)*int64(args.RoundsPerEpoch.Value) + numRoundsAfterSupernovaEnableEpoch
+		supernovaActivationRound, _ := strconv.ParseInt(configs.RoundConfig.RoundActivations[string(common.SupernovaRoundFlag)].Round, 10, 64)
+		if supernovaActivationRound > correctRoundActivationForSupernova {
+			diffRounds := supernovaActivationRound - correctRoundActivationForSupernova
+			diffIsGreaterThanAnEpoch := diffRounds > int64(args.RoundsPerEpoch.Value)-numRoundsAfterSupernovaEnableEpoch
```

Comment:
isn't this a supernova epoch?
```suggestion
			diffIsGreaterThanAnEpoch := diffRounds > int64(args.SupernovaRoundsPerEpoch.Value)-numRoundsAfterSupernovaEnableEpoch
```